llvm/lib/Target/AArch64/AArch64TargetTransformInfo.cpp

```diff
@@ -1112,6 +1112,19 @@ struct SVEIntrinsicInfo {
     return *this;
   }
 
+  bool hasMatchingIROpode() const { return IROpcode != 0; }
+
+  unsigned getMatchingIROpode() const {
+    assert(hasMatchingIROpode() && "Propery not set!");
+    return IROpcode;
+  }
+
+  SVEIntrinsicInfo &setMatchingIROpcode(unsigned Opcode) {
+    assert(!hasMatchingIROpode() && "Cannot set property twice!");
```

Comment:
nit: I'm not sure I see the value of this assert, as I could imagine a use-case for someone wanting override a previously set opcode?

Reply:
I consider that a bug.  The intent is for an intrinsic's `SVEIntrinsicInfo` to be constant and I have a hope to one day make this static rather than having to recompute it as is done today.